online/CMakeLists.txt

```
@@ -0,0 +1,70 @@
+ ################################################################################
+ #    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    #
+ #                                                                              #
+ #              This software is distributed under the terms of the             #
+ #              GNU Lesser General Public Licence (LGPL) version 3,             #
+ #                  copied verbatim in the file "LICENSE"                       #
+ ################################################################################
+
+if(BUILD_MBS)
+  add_subdirectory(MbsAPI)
+endif()
+
+set(target Online)
+
+set(sources
+  source/FairOnlineSource.cxx
+  source/FairUnpack.cxx
+
+  steer/FairRunOnline.cxx
+)
+
+if(BUILD_MBS)
+  list(APPEND sources
+    source/exitCli.c
+    source/FairLmdSource.cxx
+    source/FairMbsSource.cxx
+    source/FairMbsStreamSource.cxx
+    source/FairRemoteSource.cxx
+    source/MRevBuffer.cxx
+    source/rclose.c
+    source/swaplw.c
+  )
+endif()
+fair_change_extensions_if_exists(.cxx .h FILES "${sources}" OUTVAR headers)
+
+add_library(${target} SHARED ${sources} ${headers})
+fairroot_library_settings(${target})
+
+target_include_directories(${target} PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/steer>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
+target_link_libraries(${target} PUBLIC
+  FairRoot::FairTools
+  FairRoot::Base
+  FairRoot::ParBase
+  $<$<BOOL:${BUILD_MBS}>:FairRoot::MbsAPI>
+
+  ROOT::RHTTP # THttpServer
+  ROOT::Core
+  ROOT::Geom
+  ROOT::Net # TSocket
+)
+
+target_compile_definitions(${target} PRIVATE
+  $<$<BOOL:${BUILD_MBS}>:BUILD_MBS>
+  $<$<BOOL:${ROOT_gdml_FOUND}>:ROOT_HAS_GDML>
+  # $<$<PLATFORM_ID:Linux>:Linux>
+)
+
+fairroot_target_root_dictionary(${target}
+  HEADERS ${headers}
+  LINKDEF OnlineLinkDef.h
```

```suggestion
  LINKDEF LinkDef.h
```

As there is only one LinkDef in this directory, could we start naming it just `LinkDef.h`? I think, ROOT lately adopted this scheme as well.

---

Should I also do the same in the base directory?

---

If you like to? Yeah, sure.